build/bin/network-validator.go

```diff
@@ -108,17 +119,56 @@ func TestEndpoints(config reachabilityConfig) {
 	os.Exit(0)
 }
 
-func ValidateReachability(host string, port int, tlsDisabled bool) error {
+func embedProxyCertificate(cacertFile string) (*x509.CertPool, error) {
+	// Get the SystemCertPool, continue with an empty pool on error
+	rootCAs, _ := x509.SystemCertPool()
+	if rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	if cacertFile == "" {
+		return rootCAs, nil
```

Comment:
MB it's better to just `return nil, nil`,  because the host's root CA set would automatically be assigned: https://github.com/openshift/osd-network-verifier/pull/102/files#diff-28241c8afa458c52b0fa61a1db93fad65475845e9d58b1887cb48707385fc8acR169